### ISSUE TITLE
dotnet test fails with error "Cannot get symbol ubrk_close_68 from libicui18n" with AppLocalIcu

### ISSUE THREAD

**Issue by martincostello** [dotnet/runtime#57942]
### Description

When running `dotnet test` with a reference to the `Microsoft.ICU.ICU4C.Runtime` NuGet package version `68.2.0.6` and the `<RuntimeHostConfigurationOption Include="System.Globalization.AppLocalIcu" Value="68.2" />` item in the `.csproj` file with version `6.0.100-rc.1.21420.39` of the .NET SDK, the test process fails to run with the error below:

```sh
Testhost process exited with error: Cannot get symbol ubrk_close_68 from libicui18n
Error: 127
. Please check the diagnostic logs for more information.
```

To replicate the issue, clone https://github.com/martincostello/dotnet-test-icu-repro and run `dotnet test` with version `6.0.100-rc.1.21420.39` of the .NET SDK.

### Configuration

```sh
dotnet --info
.NET SDK (reflecting any global.json):
 Version:   6.0.100-rc.1.21420.39
 Commit:    742b676ffb

Runtime Environment:
 OS Name:     Windows
 OS Version:  10.0.19043
 OS Platform: Windows
 RID:         win10-x64
 Base Path:   C:\Program Files\dotnet\sdk\6.0.100-rc.1.21420.39\

Host (useful for support):
  Version: 6.0.0-rc.1.21420.15
  Commit:  236e490d4f
```

### Regression?

Yes, the issue was not present at least as late as .NET SDK version `6.0.100-rc.1.21406.22`.

**Comment by martincostello** [comment]
Through trial and error I've found that the change that caused this is somewhere between `6.0.100-rc.1.21418.13` and `6.0.100-rc.1.21418.17`.

**Comment by tarekgh** [comment]
@martincostello thanks for reporting it. I'll look.

**Comment by martincostello** [comment]
Might have been caused by #57078.

**Comment by tarekgh** [comment]
> Might have been caused by #57078.

Yes, but it is not clear why it fails to find ubrk_close which is supported since ICU version 2.0.

**Comment by martincostello** [comment]
The declaration is `PER_FUNCTION_BLOCK(ubrk_close, libicui18n, true) \`, but [this document](https://docs.microsoft.com/en-us/uwp/win32-and-com/win32-extension-apis) (search for `ubrk_close`) seems to suggest it should be in `icuuc`.

> ubrk_close | Introduced into icuuc.dll in 10.0.16299.

Maybe this fix it to change it to ``PER_FUNCTION_BLOCK(ubrk_close, icuuc, true) \``?

https://github.com/dotnet/runtime/blob/b8a5a4aaca93a409c09ad6cbb95e18b2626040ec/src/libraries/Native/Unix/System.Globalization.Native/pal_icushim_internal.h#L76

**Comment by martincostello** [comment]
Might be the same for `ubrk_openRules` too.

https://github.com/dotnet/runtime/blob/b8a5a4aaca93a409c09ad6cbb95e18b2626040ec/src/libraries/Native/Unix/System.Globalization.Native/pal_icushim_internal.h#L77

> ubrk_openRules | Introduced into icuuc.dll in 10.0.16299.

`usearch_getBreakIterator` seems to roughly match that doc:

https://github.com/dotnet/runtime/blob/b8a5a4aaca93a409c09ad6cbb95e18b2626040ec/src/libraries/Native/Unix/System.Globalization.Native/pal_icushim_internal.h#L160

> usearch_getBreakIterator | Introduced into icuin.dll in 10.0.16299.

**Comment by tarekgh** [comment]
Yes the whole breaker APIs should belong to the same module. The weird things is not failing on Linux which supposes the same structure. anyway, I'll look more.

**Comment by martincostello** [comment]
Confirmed fixed with version `6.0.100-rc.1.21424.38` of the SDK. Thanks for the quick turn around 👍

**Comment by tarekgh** [comment]
Thanks lot @martincostello
